core/integration/suite_test.go

```diff
@@ -246,61 +247,167 @@ func daggerCliPath(t *testing.T) string {
 	return cliPath
 }
 
-type DaggerDoCmd struct {
-	ProjectLocalPath string
-	TestGitProject   bool
-	Config           string
-	OutputPath       string
-	Target           string
-	Flags            map[string]string
-}
-
-func (do DaggerDoCmd) Run(ctx context.Context, t *testing.T, c *dagger.Client) (*dagger.Container, error) {
+func daggerCliFile(t *testing.T, c *dagger.Client) *dagger.File {
 	t.Helper()
 	cliPath := daggerCliPath(t)
 	parentDir := filepath.Dir(cliPath)
 	baseName := filepath.Base(cliPath)
-	daggerCli := c.Host().Directory(parentDir, dagger.HostDirectoryOpts{Include: []string{baseName}}).File(baseName)
-	cliBinPath := "/bin/dagger"
+	return c.Host().Directory(parentDir, dagger.HostDirectoryOpts{Include: []string{baseName}}).File(baseName)
```

Comment:
Beautiful, thank you @TomChv! Updated

Reply:
Actually I just noticed a tiny issue that's causing the tests to fail; the new API doesn't set include flags to ensure only the file gets synced, so when the tests run in CI (where the dagger CLI is put at `/.dagger-cli` in a container), they fail because they're trying to sync the whole `/`.

Sending a separate PR to fix this, it's very quick, but will revert this change in the meantime.

Reply:
Okay [fixed](https://github.com/dagger/dagger/pull/5345) and rebased, so went back to the more elegant code here now